packages/zoe/test/unitTests/contractSupport/test-zoeHelpers.js

```diff
@@ -480,174 +302,192 @@ test.skip('ZoeHelpers isOfferSafe', t => {
-    mockZCFBuilder.addBrand(simoleanR);
-    mockZCFBuilder.addAllocation(leftOfferHandle, { Asset: moola(10) });
-    mockZCFBuilder.addAllocation(rightOfferHandle, { Money: simoleans(6) });
-    mockZCFBuilder.addOffer(leftOfferHandle, {
-      proposal: {
-        give: { Asset: moola(10) },
-        want: { Bid: simoleans(4) },
-        exit: { onDemand: null },
-      },
-    });
-    mockZCFBuilder.addOffer(rightOfferHandle, {
-      proposal: {
-        give: { Money: simoleans(6) },
-        want: { Items: moola(7) },
-        exit: { onDemand: null },
-      },
-    });
-    const mockZCF = mockZCFBuilder.build();
-    const { trade } = makeZoeHelpers(mockZCF);
-    t.doesNotThrow(() =>
+test('ZoeHelpers satisfies simple proposal', t => {
+  const { moolaR, moola, simoleans } = setup();
+  const fakeZcfSeat = harden({
+    getCurrentAllocation: () => harden({ Asset: moola(10) }),
+    getProposal: () => harden({ want: { Desire: moola(30) } }),
+  });
+  const mockZCFBuilder = makeMockTradingZcfBuilder();
+  mockZCFBuilder.addBrand(moolaR);
+  const mockZCF = mockZCFBuilder.build();
+  t.falsy(
+    satisfies(mockZCF, fakeZcfSeat, { Desire: moola(3) }),
+    `giving less than specified to a proposal is not satisifying`,
+  );
+  t.falsy(
+    satisfies(mockZCF, fakeZcfSeat, { Gift: moola(3) }),
+    `giving other than what's specified to a proposal is not satisifying`,
+  );
+  t.truthy(
+    satisfies(mockZCF, fakeZcfSeat, { Desire: moola(30) }),
+    `giving exactly what's proposed is satisifying`,
+  );
+  t.truthy(
+    satisfies(mockZCF, fakeZcfSeat, { Desire: moola(30), Gift: simoleans(1) }),
+    `giving extras beyond what's proposed is satisifying`,
+  );
+  t.truthy(
+    satisfies(mockZCF, fakeZcfSeat, { Desire: moola(40) }),
+    `giving more than what's proposed is satisifying`,
+  );
+});
+
+test('ZoeHelpers satisfies() with give', t => {
+  const { moolaR, moola, bucks, bucksR, simoleans, simoleanR } = setup();
+  const fakeZcfSeat = harden({
+    getCurrentAllocation: () => harden({ Charge: moola(30) }),
+    getProposal: () => harden({ give: { Charge: moola(30) }, want: { Desire: bucks(5) } }),
+  });
+  const mockZCFBuilder = makeMockTradingZcfBuilder();
+  mockZCFBuilder.addBrand(moolaR);
+  mockZCFBuilder.addBrand(bucksR);
+  mockZCFBuilder.addBrand(simoleanR);
+  const mockZCF = mockZCFBuilder.build();
+  t.falsy(
+    satisfies(mockZCF, fakeZcfSeat, { Charge: moola(0), Desire: bucks(1) }),
+    `giving neither give nor want to a proposal is not satisifying`,
+  );
+  t.falsy(
+    satisfies(mockZCF, fakeZcfSeat, { Charge: moola(3) }),
+    `giving less than what's specified to a proposal is not satisifying`,
+  );
+  t.truthy(
+    satisfies(mockZCF, fakeZcfSeat, { Charge: moola(0), Desire: bucks(40) }),
+    `giving more than what's wanted is satisifying`,
+  );
+  t.truthy(
+    satisfies(mockZCF, fakeZcfSeat, { Desire: bucks(40), Charge: moola(3) } ),
+    `giving what's wanted makes it possible to reduce give`,
```

Comment:
Hmm, I feel like something is missing here. `satisfies` is only about the `want`, so the `give` could be a full refund or nothing at all and `satisfies` would be true as long as the `want` is satisfied. The language about `given less than what's specified to a proposal is not satisfying` seems to indicate that the `give` makes a difference. I might be misinterpreting it though